components/server/src/auth/authenticator.ts

```diff
@@ -72,18 +72,59 @@ export class Authenticator {
         });
     }
     protected async authCallbackHandler(req: express.Request, res: express.Response, next: express.NextFunction) {
-        if (req.url.startsWith("/auth/")) {
-            const hostContexts = this.hostContextProvider.getAll();
-            for (const { authProvider } of hostContexts) {
-                const authCallbackPath = authProvider.authCallbackPath;
-                if (req.url.startsWith(authCallbackPath)) {
```

Comment:
no longer relying on the `authCallbackPath` being configured on the authProvider.

reasons are:
1. using a static endpoint for callbacks (`/auth/callback`, and a compatibility pattern `/auth/<host_like>/callback`)
2. encoding the host into the state param, which is required to be sent back as per protocol (this is already done before this PR)